src/project.js

```diff
@@ -152,3 +153,83 @@ export function setTimezone(projectId, timezone) {
     }).then(parseJSON);
 }
 
+const isProjectCreationDone = (project) => {
+    const projectState = project.content.state;
+
+    return projectState === 'ENABLED' ||
+        projectState === 'DELETED';
+};
+
+const waitUntilProjectEnabled = (response, options = {}) => {
+    const { pollStep = 5000 } = options;
+    const projectUri = response.uri;
+
+    return new Promise((resolve, reject) => {
+        AsyncPolling(function(end) { // eslint-disable-line func-names
```

Comment:
I know it's still far from finished, but that polling code seems too complicated.

Assuming using bluebird, we could do something like:
```js
const Promise = require('bluebird');

function poll() {
    return getProject().then((project) => {
        if (isProjectCreated(project)) {
            return Promise.resolve();
        }

        return Promise.delay(5000).then(poll);
    });
}
```